Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -938,52 +938,44 @@ bool MMU::IsOptimizableRAMAddress(const u32 address, const u32 access_size) cons
 }
 
 template <XCheckTLBFlag flag>
-bool MMU::IsRAMAddress(u32 address, bool translate)
+bool MMU::IsTranslatedRAMAddress(u32 address)
 {
-  if (translate)
+  if (auto translate_address = TranslateAddress<flag>(address); translate_address.Success())
   {
-    auto translate_address = TranslateAddress<flag>(address);
-    if (!translate_address.Success())
-      return false;
     address = translate_address.address;
-  }
-
-  u32 segment = address >> 28;
-  if (m_memory.GetRAM() && segment == 0x0 && (address & 0x0FFFFFFF) < m_memory.GetRamSizeReal())
-  {
-    return true;
-  }
-  else if (m_memory.GetEXRAM() && segment == 0x1 &&
-           (address & 0x0FFFFFFF) < m_memory.GetExRamSizeReal())
-  {
-    return true;
-  }
-  else if (m_memory.GetFakeVMEM() && ((address & 0xFE000000) == 0x7E000000))
-  {
-    return true;
-  }
-  else if (m_memory.GetL1Cache() && segment == 0xE &&
-           (address < (0xE0000000 + m_memory.GetL1CacheSize())))
-  {
-    return true;
+    return IsRAMAddress<flag>(translate_address.address);
   }
   return false;
 }
 
+template <XCheckTLBFlag _>
```

Comment:
If you're not using the template argument, this function shouldn't be templated.